plugins/civmodcore-paper/src/main/java/vg/civcraft/mc/civmodcore/config/ConfigHelper.java

```diff
@@ -269,14 +273,117 @@ public static ItemMap parseItemMapDirectly(@Nullable final ConfigurationSection
 				}
 
 			}
-			toAdd.setItemMeta(meta);
-			if (current.contains("nbt")) {
-				toAdd = ItemMap.enrichWithNBT(toAdd, 1, current.getConfigurationSection("nbt").getValues(true));
+			if (current.contains("max_stack_size")) {
+				meta.setMaxStackSize(current.getInt("max_stack_size"));
+			}
+			if (current.contains("damage")) {
+				((Damageable) meta).setDamage(current.getInt("damage"));
+			}
+			if (current.contains("rarity")) {
+				meta.setRarity(ItemRarity.valueOf(current.getString("rarity")));
+			}
+			if (current.contains("custom_model_data")) {
+				meta.setCustomModelData(current.getInt("custom_model_data"));
+			}
+			if (current.contains("hide_tooltip")) {
+				meta.setHideTooltip(current.getBoolean("hide_tooltip"));
+			}
+			if (current.contains("repair_cost")) {
+				net.minecraft.world.item.ItemStack nmsItem = ItemUtils.getNMSItemStack(toAdd);
+				nmsItem.set(DataComponents.REPAIR_COST, current.getInt("repair_cost"));
+
+				toAdd = nmsItem.getBukkitStack();
```

Comment:
From what I remember, ever ItemMeta implements Repairable, so you should be able to do what you did above with damage. Or you could do an [instaceof pattern match](https://openjdk.org/jeps/394).